app/scripts/modules/core/src/task/monitor/TaskMonitorWrapper.tsx

```diff
@@ -1,5 +1,66 @@
+import React from 'react';
+
 import { TaskMonitor } from 'core/task';
+import { TaskMonitorStatus } from './TaskMonitorStatus';
+import { TaskMonitorError } from './TaskMonitorError';
+import { CSSTransition } from 'react-transition-group';
 
 export interface ITaskMonitorProps {
   monitor: TaskMonitor;
 }
+
+export const TaskMonitorWrapper = ({ monitor }: ITaskMonitorProps) => {
+  const { useState, useEffect } = React;
+
+  /***
+   * Super hacky way to deal with props not updating when the task monitor updates
+   */
+  const [, setIteration] = useState(Date.now());
```

Comment:
Recommend creating a `useForceUpdate` hook that encapsulates this wonkyness

something like: 
```js
export function useForceUpdate() {
  const [, setToggle] = useState(false);
  return React.useCallback(() => setToggle(toggle => !toggle), []);
}
```

then the usage becomes clearer and less wonky:

```js
const forceUpdate = useForceUpdate();
useEffect(() => {
    const subscription = monitor.statusUpdatedStream.subscribe(() => {
      forceUpdate();
    });
    return () => subscription.unsubscribe();
  }, [monitor]);
```

Reply:
What about this:
```javascript
export const useForceRerender = (observable: Observable<any>) => {
  const [, setTick] = useState(Date.now());

  useEffect(() => {
    const subscription = observable.subscribe(() => setTick(Date.now()));
    return () => subscription.unsubscribe();
  }, [observable]);
};
```
and then the usage is very succinct:
```javascript
useForceRerender(monitor.statusUpdatedStream);
// no more useState, useEffect
```

or do you feel like that is too opinionated and bespoke?

Reply:
I think it's maybe overly opinionated.  I suspect `forceUpdate` is more likely to be reused.

Reply:
Sounds good - I will do a forceUpdate hook